05-deploying_shiny_app.Rmd

```diff
@@ -1,279 +1,239 @@
 
-In 'Repository URL', enter the URL of the repository you just created.  This needs to be the SSH URL (starting with `git@github`), **NOT** the HTTPS url (which starts with `https`). Therefore, it is in the following format:
-
-```
-git@github.com:moj-analytical-services/your_name_goes_here.git
-```
-
-R Studio will create a new project folder with all the template files in.
-
-You can now start building your Shiny app by editing `server.R` and `ui.R` files.
-
-#### Further notes if you're having trouble finding your new repo's url
-
-If you navigate to your new repository's home page (which will have a url in the form `https://github.com/moj-analytical-services/your_name_goes_here`), you can use the following buttons to access this url (make sure you click the 'ssh' button):
-
-![](images/shiny/use_ssh.PNG)
-
-![](images/shiny/ssh_url.PNG)
+### Manage dependencies
 
+Most apps will have dependencies on various third-party packages. Changes to these packages may not always be backwards-compatible. To avoid compatibility issues and ensure reproducible outputs, it is necessary to use a package management system, such as packrat or conda.
 
-### Work on your app
+If using packrat, ensure that it is enabled for your project in RStudio.
 
-You can now work on your Shiny app in R Studio as normal.  As you work, commit your changes to Github using your chosen Github workflow.
+To enable packrat, select 'Tools' > 'Project Options...' > ' Packrat' > 'Use packrat with this project'.
 
-### Snapshot dependencies using packrat
+When packrat is enabled, run `packrat::snapshot()` to generate a list of packages used in the project, their sources and their current versions.
 
-Your Shiny app will have dependencies on various third-party R packages (e.g. `dplyr`).  These packages change through time, and sometimes these changes are not backwards compatible.  When we deploy your Shiny app, it is therefore essential to include a manifest of all of the package dependencies and their specific versions.
+You may also wish to run `packrat::clean()` to remove unused packages from the list.
 
-The R package`packrat` automates this process.  Before deploying your app, you need to call `packrat::snapshot()` to automatically create this manifest.  You may want to call `packrat::clean()` to remove any dependencies that are no longer in use.
+The list is stored in a file called `packrat/packrat.lock`. You must ensure that you have committed this file to GitHub before deploying your app.
 
-This creates a file called `packrat/packrat.lock`.  You need to ensure you have committed this to the Github repository prior to deployment.
+### Set access permissions
 
-![](images/shiny/github_packrat_snapshot.gif)
+You can set some access permissions for your app in the `deploy.json` file.
```

Comment:
This is the first time you've mentioned deploy.json. Somewhere before this I think we need the sentence `Include a deploy.json file so that Concourse will automatically build and deploy your app, so that it is running and customers can access it. For more about this see: Build and deploy guidance`

Reply:
I'm not sure linking to the build and deploy guidance is actually that useful as it just send users back to the section of the guidance for that specific type of deployment.

Reply:
Have added a bit more on `deploy.json` in c1f09c1.

Reply:
I like it 👍